src/System.Text.Primitives/System/Text/Parsing/Signed.cs

```diff
@@ -11,7 +11,12 @@ public static partial class PrimitiveParser
     {
         public static bool TryParseSByte(ReadOnlySpan<byte> text, out sbyte value, out int bytesConsumed, EncodingData encoding = default(EncodingData), TextFormat format = default(TextFormat))
         {
-            if (format.HasPrecision)
+            if (encoding == default(EncodingData))
+            {
+                encoding = EncodingData.InvariantUtf8;
+            }
+
+            if (!format.IsDefault && format.HasPrecision)
```

Comment:
For the bug fix, I think it's ok, but I think it would be good to open an issue and design the precondition contract for these routines and implement them at the top of the routine. Otherwise: a) it's not clear what combination of arguments is valid, b) we are checking some things (e.g. IsDefault) more than one time.